File: ruby-client/lib/rakam_client/api/retention_api.rb
require "uri"

module RakamClient
  class RetentionApi
    attr_accessor :api_client

    def initialize(api_client = ApiClient.default)
      @api_client = api_client
    end

    # Execute query
    # 
    # @param retention_query 
    # @param [Hash] opts the optional parameters
    # @return [QueryResult]
    def retention_analyzer_execute(retention_query, opts = {})
      data, status_code, headers = retention_analyzer_execute_with_http_info(retention_query, opts)
      return data
    end

    # Execute query
    # 
    # @param retention_query 
    # @param [Hash] opts the optional parameters
    # @return [Array<(QueryResult, Fixnum, Hash)>] QueryResult data, response status code and response headers
    def retention_analyzer_execute_with_http_info(retention_query, opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: RetentionApi#retention_analyzer_execute ..."
      end
      
      # verify the required parameter 'retention_query' is set
      fail "Missing the required parameter 'retention_query' when calling retention_analyzer_execute" if retention_query.nil?
      
      # resource path
      path = "/retention/analyze".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = @api_client.object_to_http_body(retention_query)
      

      auth_names = ['read_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'QueryResult')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: RetentionApi#retention_analyzer_execute\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end
  end
end

File: java-client/src/main/java/org/rakam/client/api/RetentionApi.java
package org.rakam.client.api;

import com.sun.jersey.api.client.GenericType;

import org.rakam.ApiException;
import org.rakam.ApiClient;
import org.rakam.Configuration;
import org.rakam.Pair;

import org.rakam.client.model.QueryResult;
import org.rakam.client.model.RetentionQuery;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-05-06T02:48:07.215+03:00")
public class RetentionApi {
  private ApiClient apiClient;

  public RetentionApi() {
    this(Configuration.getDefaultApiClient());
  }

  public RetentionApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Execute query
   * 
   * @param retentionQuery 
   * @return QueryResult
   */
  public QueryResult retentionAnalyzerExecute(RetentionQuery retentionQuery) throws ApiException {
    Object postBody = retentionQuery;
    
     // verify the required parameter 'retentionQuery' is set
     if (retentionQuery == null) {
        throw new ApiException(400, "Missing the required parameter 'retentionQuery' when calling retentionAnalyzerExecute");
     }
     
    // create path and map variables
    String path = "/retention/analyze".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "read_key" };

    
    GenericType<QueryResult> returnType = new GenericType<QueryResult>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
}
